### PR TITLE
Update to pack 0.15 and add debug support for CNB Platform API 0.4

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes: #4796 

**Description**
Update to pack 0.15.0, and update `skaffold debug` to support the Cloud Native Buildpacks _Platform API 0.4_, the default for the CNB `pack` ≥ 0.13.0 and lifecycle ≥ 0.9.  Platform API 0.4 changes how _processes_ are requested in an image:
  - prior to 0.4, the entrypoint points to a launcher binary in `/cnb/lifecycle/launcher`, and the process is set via a `CNB_PROCESS_TYPE` environment variable (defaulting to `web` if not present)
  - with 0.4, the processes have corresponding executables in `/cnb/process/xxx` which are symlinks to `/cnb/lifecycle/launcher`.  When executed, they cause the corresponding process to be run.  The default process is the entrypoint (e.g., `/cnb/process/web`).

The Platform API version is set via the new `CNB_PLATFORM_API` environment variable.

**User facing changes (remove if N/A)**

(This doesn't actually have any user-facing changes as to this point we still link with pack 0.12 and so create images with CNB Platform API 0.3.)

### CODE REVIEW

File: pkg/skaffold/debug/cnb.go [hunk 1]
```diff
@@ -97,58 +148,51 @@ func updateForCNBImage(container *v1.Container, ic imageConfiguration, transform
...
+			result = append(result, clArgs...)
+			return result
+		}
 	}
 
 	if len(ic.arguments) == 0 {
-		// indicates an image mis-configuration as we should have resolved the the
-		// CNB_PROCESS_TYPE (if specified) or `web`.
-		logrus.Warnf("no CNB launch found for %s/%s", ic.artifact, processType)
+		logrus.Warnf("no CNB launch found for %s", ic.artifact)
```
Comment (tejal29) [inline pkg/skaffold/debug/cnb.go:195]: would it possible to debug a container if CNB launch is not found?

**Review by tejal29 — APPROVED** [review]
LGTM.

File: pkg/skaffold/debug/cnb.go [hunk 2]
```diff
@@ -19,45 +19,89 @@ package debug
...
+//   3. _shell scripts_: the user can provide a shell script as the first argument and any
+//      remaining arguments are available as positional arguments like _script processes_.
 //
-// A key point is that the script-style launches allow support references to environment variables.
-// So we find the command line to be executed, whether that is a script or a direct, and turn it into
-// a normal command-line.  But we also return a rewriter to transform the the command-line back to
-// the original form.
+// Script-style launches support referencing environment variables since they are expanded by the shell.
+//
+// Configuring the launch depends on the CNB Platform API version being used, which is determined by
```
Comment (tejal29) [inline pkg/skaffold/debug/cnb.go:83]: Thanks for the details, it was a little bit easier to understand the code.

Comment (briandealwis) [reply pkg/skaffold/debug/cnb.go:83]: Going through this was painful :-)

File: pkg/skaffold/debug/cnb.go (hunk 1 above)
Comment (briandealwis) [reply pkg/skaffold/debug/cnb.go:195]: No.  At this point we have an CNB launcher entrypoint (either `/cnb/lifecycle/launcher` or `/cnb/process/xxx`), but no corresponding process found.  So we don't have a command-line should be executed.

Basically we'll return, the debug transforms won't be able to figure anything out, and the container / pod will remain un-touched.  And the execution of the container will then fail as no process definition will be found.

**Comment by briandealwis** [comment]
Marking TestDevPortForwardDeletePod to be skipped #4733.

**Review by nkubala — APPROVED** [review]
this LGTM, its essentially an improved version of similar changes I made in https://github.com/GoogleContainerTools/skaffold/pull/5038. thanks for the detailed comments @briandealwis

**Pull request merged**